app/src/main/java/org/thoughtcrime/securesms/util/ImageCompressionUtil.java

```diff
@@ -0,0 +1,130 @@
+package org.thoughtcrime.securesms.util;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+
+import androidx.annotation.IntRange;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.WorkerThread;
+
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+
+import org.thoughtcrime.securesms.mms.GlideApp;
+
+import java.io.ByteArrayOutputStream;
+import java.util.concurrent.ExecutionException;
+
+public final class ImageCompressionUtil {
+
+  private ImageCompressionUtil () {}
+
+  /**
+   * A result satisfying the provided constraints, or null if they could not be met.
+   */
+  @WorkerThread
+  public static @Nullable Result compressWithinConstraints(@NonNull Context context,
+                                                           @NonNull String mimeType,
+                                                           @NonNull Object glideModel,
+                                                           int maxDimension,
+                                                           int maxBytes,
+                                                           @IntRange(from = 0, to = 100) int quality)
+      throws BitmapDecodingException
+  {
+    Result result = compress(context, mimeType, glideModel, maxDimension, quality);
+
+    if (result.getData().length <= maxBytes) {
+      return result;
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Compresses the image to match the requested parameters.
+   */
+  @WorkerThread
+  public static @NonNull Result compress(@NonNull Context context,
+                                          @NonNull String mimeType,
+                                          @NonNull Object glideModel,
+                                          int maxDimension,
+                                          @IntRange(from = 0, to = 100) int quality)
+      throws BitmapDecodingException
+  {
+    Bitmap scaledBitmap;
+
+    try {
+      scaledBitmap = GlideApp.with(context.getApplicationContext())
+                             .asBitmap()
+                             .load(glideModel)
+                             .skipMemoryCache(true)
+                             .diskCacheStrategy(DiskCacheStrategy.NONE)
+                             .centerInside()
+                             .submit(maxDimension, maxDimension)
+                             .get();
+    } catch (ExecutionException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)